app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -73,31 +76,34 @@ class FxaRepo(
         return accountManager.beginAuthenticationAsync()
     }
 
-    private inner class FirefoxAccountObserver : AccountObserver {
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    inner class FirefoxAccountObserver : AccountObserver {
+        /**
+         * The account profile is fetched asynchronously.
+         * There is no way to transition from [NOT_AUTHENTICATED] directly to [AUTHENTICATED_WITH_PROFILE];
```

Comment:
Is this true? What happens if the user is already authenticated? Is the profile saved to disk?

Reply:
Looks like account is saved but profile is not. So even if account is saved, the profile needs to be fetched.